core/src/main/scala/scalaz/MaybeT.scala

```diff
@@ -9,15 +9,25 @@ final case class MaybeT[F[_], A](run: F[Maybe[A]]) {
 
   def map[B](f: A => B)(implicit F: Functor[F]): MaybeT[F, B] = new MaybeT[F, B](mapO(_ map f))
 
+  def mapF[B](f: A => F[B])(implicit F: Monad[F]): MaybeT[F, B] = new MaybeT[F, B](
+    F.bind(self.run) {
+        case Empty() => F.point(empty[B])
+        case Just(z) => F.map(f(z))(b => just(b))
+      }
+    )
+
   def mapT[G[_], B](f: F[Maybe[A]] => G[Maybe[B]]): MaybeT[G, B] =
     MaybeT(f(run))
 
   def flatMap[B](f: A => MaybeT[F, B])(implicit F: Monad[F]): MaybeT[F, B] = new MaybeT[F, B](
     F.bind(self.run)(_.cata(f(_).run, F.point(empty)))
   )
 
-  def flatMapF[B](f: A => F[B])(implicit F: Monad[F]): MaybeT[F, B] = new MaybeT[F, B](
-    F.bind(self.run)(_.cata((a => F.map(f(a))(just)), F.point(empty)))
+  def flatMapF[B](f: A => F[Maybe[B]])(implicit F: Monad[F]): MaybeT[F, B] = new MaybeT[F, B](
```

Comment:
I think this is not binary compatible as well https://github.com/lightbend/migration-manager/issues/40

Reply:
the workaround is to add it to syntax that is opt-in.